setup.cfg

```diff
@@ -28,5 +28,18 @@ max-line-length = 95
 ignore = E116,E241,E251
 exclude = .git,.tox,.venv,tests/*,build/*,sphinx/search/*,sphinx/pycode/pgen2/*,doc/ext/example*.py
 
+[mypy]
+python_version = 2.7
+show_column_numbers = True
+show_error_context = False
+ignore_missing_imports = True
+follow_imports = skip
+cache_dir = /dev/null
```

Comment:
I feel this is harmful. why do you disable caches?

Reply:
In addition, I think this brings troubles to windows environment.

Reply:
> I feel this is harmful. why do you disable caches?

This was causing me issues when working on #4097. Whenever I added a line like [this](https://github.com/sphinx-doc/sphinx/pull/4097/commits/3559577d5ef37211a1e3cd9222d560a0541d6328#diff-c3a9ba1cd0902193425499e0cf7827b6R288) I would get error messages. All this does is improve performance a small bit, so I figured I could accept the cost.